FOCUS #76: Create Datetime Requirements attribute (#123)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/attributes/datetime_requirements.md

> # Datetime Requirements

Reviewer: ```suggestion
# Date/Time Requirements
```

Author: suggested change applied

> Billing data columns that provide date and time information (datetime related columns) conforming to specified rules and formatting requirements ensure clarity, accuracy and ease of interpretation for both humans and systems.

Reviewer: nit: We shouldn't need to say "billing data" since all date/time columns would follow this, right? Also, the second datetime reference seems redundant.
```suggestion
Columns that provide date and time information conforming to specified rules and formatting requirements ensure clarity, accuracy, and ease of interpretation for both humans and systems.
```

Author: suggested changes applied

> All datetime related columns, defined in the FOCUS specification, MUST be represented in UTC and follow the ISO 8601 aligned formatting requirements listed below. Provider generated datetime related columns SHOULD adopt the same format requirements over time.

Reviewer 2: Are we sure on providers adopting this format?  That seems like a big assumption to be giving people.

Author: I'll reduce this provider related expectation to _use UTC and align with ISO 8601_.

Author: provider-related expectations reworded to: "Provider generated date/time related columns SHOULD also be represented in UTC and conform to the ISO 8601 standard."

Reviewer 3: I think we should have kept it as it was previously. Maybe something to revisit later? By saying should 'SHOULD', it was a recommendation not a requirement based on the fact that practitioners would benefit from some standardization of dates/times.

> All datetime related columns, defined in the FOCUS specification, MUST be represented in UTC and follow the ISO 8601 aligned formatting requirements listed below. Provider generated datetime related columns SHOULD adopt the same format requirements over time.

Reviewer: "datetime" is technically not an English word as far as I know. I believe it's just tech jargon.
No need to say "over time". 
```suggestion
All date/time related columns, defined in the FOCUS specification, MUST be represented in UTC and follow the ISO 8601 aligned formatting requirements listed below. Provider generated date/time related columns SHOULD adopt the same format requirements.
```

Author: suggested changes applied

## Attribute ID

DatetimeRequirements

## Attribute name

Datetime Requirements

## Description

Rules and formatting requirements for datetime related columns appearing in billing data.

## Requirements

* Datetime values MUST be in UTC (Coordinated Universal Time) to avoid ambiguity and ensure consistency across different time zones.
* Datetime values format MUST be aligned with ISO 8601 standard, which provides a globally recognized format for representing dates and times (see [ISO 8601-1:2019](https://www.iso.org/standard/70907.html) governing document for details).
* Values providing information about a specific moment in time MUST be represented in the extended ISO 8601 format with time zone offset ('YYYY-MM-DDTHH:mm:ssZ') and conform to the following guidelines:
  * Include the date and time components, separated with the letter 'T'

>   * Use two-digit hours (HH), minutes (mm), and seconds (ss).

Reviewer: I believe seconds are optional. Should we remove them or call them out as being optional?

Author: As agreed at the squad 2 meeting, for v5.0, only the second level precision format was retained in the specification. note related to other levels of precision (minutes, milliseconds, or microseconds) added to supporting content. We plan to create an additional issue on this topic.
